### PR TITLE
[basics/quickstart_tutorial.py] call model.trian() before training

### PR DESCRIPTION
**Pull request by 183amir**
Adds a `model.train()` call before training.
Because the `model.eval()` is called inside the `test()` function, the model trains in eval mode after the first eopch.

Even though `model.train()` has no effect here because the architecture does not contain batch norm or dropout layers,
A lot of people (myself included) copy code from this tutorial and calling `model.train()` before training is a good practice.

### CODE REVIEW

**Comment by 183amir** [comment]
The failed CI jobs are not related to the changes done here. It's because of a job timeout and an Internet connection error.

**Comment by guoyejun** [comment]
https://github.com/pytorch/tutorials/issues/1624, I think it is better to accept this PR sooner, people will be confused by this quick start tutorial. (a new update may needed to explain in the document)

**Comment by suraj813** [comment]
This was indeed a confusing bug, thank you @guoyejun and @183amir for calling it out and fixing it

**Review by suraj813 — APPROVED** [review]
